### ISSUE TITLE
Add WaitStrategy

### ISSUE THREAD

**Issue by MeneDev** [testcontainers/testcontainer-go#7]
WaitStrategy is essential for reliable tests. For example an integration test against a HTTP Server would fail if the container/the application in the container isn't ready to accept connections.

I'm willing to contribute on this matter. My suggestion is to closely mimic the Java API as it is proven and a similar API simplifies porting. I'm not too familiar with Go, so I don't know how idiomatic this API is for Go.

Any thoughts?

**Comment by gianarb** [comment]
Hello! Thanks for your feedback. Can you paste a code example in Java in order to understand how it handles this feature?

**Comment by MeneDev** [comment]
I experimented a bit and pushed my results here: https://github.com/MeneDev/testcontainer-go/tree/wait_strategy_proposal

As an overview:

There are 2 interfaces: `WaitStrategy` and `WaitStrategyTarget`.
`WaitStrategy` has 1 method: `WaitUntilReady` which blocks until the wait condition (defined in the implementations) is satisfied.

`WaitStrategyTarget` exposes what is needed to evaluate the wait condition: ip address and ports.

`Container` implements `WaitStrategyTarget`
`RequestContainer` has a `WaitStrategy` called `WaitingFor`

If a `WaitStrategy` is specified as input to `RunContainer`, `RunContainer` will wait for the `WaitStrategy` to complete.

As an example I implemented `hostPortWaitStrategy` (just waiting for all ports to be open) and sketched out the API of the `httpWaitStrategy`.

A typical `WaitStrategy` should:

 * have a `startupTimeout` to avoid waiting infinitely
 * expose a "constructor" method to set the fields to sensible defaults (e.g. `startupTimeout` = 60s)
 * expose a fluent builder API for all configurable properties
 * expose a "convenience method" starting with "For" in the wait package (`ForListeningPort`, `ForHttp`)

The most obvious deviation from the Java API is that most methods will take an additional `Context` parameter to control timeouts and return error values for error-handling in a go idiomatic way.

Also there is no class hierarchy which means that some things are currently duplicated. For example `hostPortWaitStrategy` as well as `httpWaitStrategy` have a `startupTimeout` property and both need to limit the context passed to `WaitUntilReady` to this timeout. I think the duplication is acceptable, but maybe there is a better way to do this in go.
